source/common/buffer/buffer_impl.cc

```diff
@@ -41,6 +41,18 @@ void OwnedImpl::add(const Instance& data) {
   }
 }
 
+void OwnedImpl::prepend(absl::string_view data) {
+  evbuffer_prepend(buffer_.get(), data.data(), data.size());
+}
+
+void OwnedImpl::prepend(Instance& data) {
+  int rc =
+      evbuffer_prepend_buffer(buffer_.get(), static_cast<LibEventInstance&>(data).buffer().get());
+  ASSERT(rc == 0);
+  data.drain(data.length());
```

Comment:
`evbuffer_prepend_buffer ` will drain `data`, do you still need calling drain here? Just ASSERT it is empty?